packages/aws-cdk-lib/aws-iam/lib/oidc-provider.ts

```diff
@@ -98,6 +98,7 @@ export interface OpenIdConnectProviderProps {
  * @see https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_oidc.html
  *
  * @resource AWS::CloudFormation::CustomResource
+ * @deprecated Use { @link OidcProvider } instead.
```

Comment:
Since the old `OpenIdConnectProvider` is still being used in EKS, I don't think we can deprecate this just yet. Apologies for the churn but I did not realize this when making the original suggestion. For the time being we will have to just support both of these. On this note as well, because the old `OpenIdConnectProvider` will not be deprecated we should probably rename the new `OidcProvider` to be `OidcProviderNative` so that there will be a distinction at least. Then in CDK v3 we will be able to remove the old `OpenIdConnectProvider` and then we can remove the `Native` from `OidcProviderNative`.